node/src/accountant/db_access_objects/failed_payable_dao.rs

```diff
@@ -0,0 +1,778 @@
+    }
+
+    #[test]
+    fn update_tx_failures_works() {
+        let home_dir =
+            ensure_node_home_directory_exists("failed_payable_dao", "update_tx_failures_works");
+        let wrapped_conn = DbInitializerReal::default()
+            .initialize(&home_dir, DbInitializationConfig::test_default())
+            .unwrap();
+        let subject = FailedPayableDaoReal::new(wrapped_conn);
+        let tx1 = FailedTxBuilder::default().hash(make_tx_hash(1)).build();
+        let tx2 = FailedTxBuilder::default().hash(make_tx_hash(2)).build();
+        subject
+            .insert_new_records(&vec![tx1.clone(), tx2.clone()])
+            .unwrap();
+        let hash_map = HashMap::from([(tx1.hash, PendingTooLong), (tx2.hash, NonceIssue)]);
+
+        let result = subject.update_tx_failures(&hash_map);
+
+        let updated_txs = subject.retrieve_txs(None);
+        assert_eq!(result, Ok(()));
+        assert_eq!(updated_txs[0].reason, PendingTooLong);
+        assert_eq!(updated_txs[1].reason, NonceIssue);
+    }
+
+    #[test]
+    fn update_tx_failures_returns_error_when_input_is_empty() {
+        let home_dir = ensure_node_home_directory_exists(
+            "failed_payable_dao",
+            "update_tx_failures_returns_error_when_input_is_empty",
+        );
+        let wrapped_conn = DbInitializerReal::default()
+            .initialize(&home_dir, DbInitializationConfig::test_default())
+            .unwrap();
+        let subject = FailedPayableDaoReal::new(wrapped_conn);
+        let existent_hash = make_tx_hash(1);
+        let tx = FailedTxBuilder::default().hash(existent_hash).build();
+        subject.insert_new_records(&vec![tx]).unwrap();
+        let hash_map = HashMap::new();
+
+        let result = subject.update_tx_failures(&hash_map);
+
+        assert_eq!(result, Err(FailedPayableDaoError::EmptyInput));
+    }
+
+    #[test]
+    fn update_tx_failures_returns_error_during_partial_execution() {
+        let home_dir = ensure_node_home_directory_exists(
+            "failed_payable_dao",
+            "update_tx_failures_returns_error_during_partial_execution",
+        );
+        let wrapped_conn = DbInitializerReal::default()
+            .initialize(&home_dir, DbInitializationConfig::test_default())
+            .unwrap();
+        let subject = FailedPayableDaoReal::new(wrapped_conn);
+        let existent_hash = make_tx_hash(1);
+        let non_existent_hash = make_tx_hash(999);
+        let tx = FailedTxBuilder::default().hash(existent_hash).build();
+        subject.insert_new_records(&vec![tx]).unwrap();
+        let hash_map = HashMap::from([
+            (existent_hash, PendingTooLong),
+            (non_existent_hash, NonceIssue),
+        ]);
+
+        let result = subject.update_tx_failures(&hash_map);
+
+        assert_eq!(
+            result,
+            Err(FailedPayableDaoError::PartialExecution(format!(
+                "Failed to update failure reason for hash {:?}",
```

Comment:
Maybe mentioning the desired failure reason...

But I think this method may change a lot. I don't want to comment it too much.